packages/@headlessui-vue/src/components/transitions/transition.ts

```diff
@@ -327,7 +329,15 @@ export let TransitionChild = defineComponent({
       } = props
 
       let ourProps = { ref: container }
-      let theirProps = rest
+      let theirProps = {
+        ...rest,
+        ...(appear && show && env.isServer
+          ? {
+              // Already apply the `enter` and `enterFrom` on the server if required
+              class: normalizeClass([rest.class, ...enterClasses, ...enterFromClasses]),
+            }
+          : {}),
```

Comment:
We do this exact thing in the React version as well, so looking good to me!

https://github.com/tailwindlabs/headlessui/blob/08c083768ff3a6f8da1547d9cbfa824d1261f0f7/packages/%40headlessui-react/src/components/transitions/transition.tsx#L417-L423